### PR TITLE
DATAUP-497: job data structure updates and  job test data edits

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Two changes in this PR:
- update the job state object validation to include the `job_id` key (change made in the last PR)
- more edits of the JS job test data; replace the valid message samples with those from the BE response data; and use constants for the job names

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 1]
```diff
@@ -254,7 +254,7 @@ define([
      * @returns {boolean} true|false
      */
     function isValidBackendJobStateObject(backendJobState) {
-        const required = ['jobState'];
+        const required = ['job_id', 'jobState'];
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:257]: this change was made in the backend in the previous PR; this adds an extra bit of validation in the frontend.

File: test/data/jobsData.js [hunk 2]
```diff
@@ -30,6 +35,20 @@ define(['common/format', 'common/jobCommMessages', 'testUtil'], (format, jcm, Te
...
+        ESTIMATING: 'JOB_ESTIMATING',
+        QUEUED: 'JOB_QUEUED',
+        TERMINATED_WHILST_QUEUED: 'JOB_TERMINATED_WHILST_QUEUED',
+        DIED_WHILST_QUEUED: 'JOB_DIED_WHILST_QUEUED',
+        RUNNING: 'JOB_RUNNING',
+        TERMINATED_WHILST_RUNNING: 'JOB_TERMINATED_WHILST_RUNNING',
+        DIED_WHILST_RUNNING: 'JOB_DIED_WHILST_RUNNING',
+        COMPLETED: 'JOB_COMPLETED',
+        UNKNOWN: 'JOB_UNKNOWN',
+        BATCH_PARENT: BATCH_ID,
```
Comment (ialarmedalien) [inline test/data/jobsData.js:49]: edited the names to be internally consistent and turned them into variables

File: test/data/jobsData.js [hunk 3]
```diff
@@ -130,7 +149,40 @@ define(['common/format', 'common/jobCommMessages', 'testUtil'], (format, jcm, Te
             },
         },
         {
-            job_id: 'job-running',
+            job_id: JOB.DIED_WHILST_QUEUED,
```
Comment (ialarmedalien) [inline test/data/jobsData.js:152]: moved this up to be next to the other jobs that got to queued but no further

File: test/data/jobsData.js [hunk 4]
```diff
@@ -372,82 +391,46 @@ define(['common/format', 'common/jobCommMessages', 'testUtil'], (format, jcm, Te
     const invalidJobStates = [
         ...invalidTypes,
         {
-            job_id: 'somejob',
+            job_id: TEST_JOB_ID,
```
Comment (ialarmedalien) [inline test/data/jobsData.js:394]: standardize to use TEST_JOB_ID for all these invalid jobs

File: test/data/jobsData.js [hunk 5]
```diff
@@ -372,82 +391,46 @@ define(['common/format', 'common/jobCommMessages', 'testUtil'], (format, jcm, Te
...
-            outputWidgetInfo: {},
-        },
-        {
-            jobState: validJobStates[2],
-            outputWidgetInfo: {
-                ping: 'pong',
-            },
-        },
-    ];
+    const validBackendJobStates = Object.values(ResponseData[jcm.MESSAGE_TYPE.STATUS]);
```
Comment (ialarmedalien) [inline test/data/jobsData.js:416]: use the data from the backend tests

File: test/data/jobsData.js [hunk 6]
```diff
@@ -372,82 +391,46 @@ define(['common/format', 'common/jobCommMessages', 'testUtil'], (format, jcm, Te
...
-            batch_id: null,
-            app_name: 'Some Crappy App',
-            app_id: 'some/crappy-app',
-            job_params: [{}],
+            jobState: validJobStates[1],
+            outputWidgetInfo: null,
         },
     ];
 
+    const validInfo = Object.values(ResponseData[jcm.MESSAGE_TYPE.INFO]);
```
Comment (ialarmedalien) [inline test/data/jobsData.js:432]: Again, the data from the backend tests

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 7]
```diff
@@ -550,23 +550,6 @@ define([
                         },
                     ],
                 },
-                {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommChannel-spec.js:553]: remove these tests for single messages and just use the tests for multiple messages

File: test/unit/spec/common/jobManager-Spec.js [hunk 8]
```diff
@@ -959,9 +959,9 @@ define([
                     return new Promise((resolve) => {
                         const message = {
                             [jcm.PARAM.JOB_ID]: this.jobId,
-                            job: { jobState: updatedJobState },
+                            job: { job_id: this.jobId, jobState: updatedJobState },
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:962]: update to make this a valid job retry message

**Review by briehl — APPROVED** [review]
Looks good!

**Pull request merged**